src/main/java/appeng/menu/SlotSemantics.java

```diff
@@ -31,6 +33,7 @@ private SlotSemantics() {
     }
 
     private static final Map<String, SlotSemantic> REGISTRY = new ConcurrentHashMap<>();
+    public static final Set<String> HIDEABLE = new HashSet<>();
```

Comment:
I still think it would be better to make it per-menu, unless that poses an issue to addons ? I wouldn't want to allow a malicious user to hide slots in the wrong menu because another menu is allowing that with the same semantic.

What about such a method in the menu class?
```java
/**
 * Return true if the client is allowed to hide slots with this semantic, false otherwise. You should return false unless you have checked that hiding the slots doesn't risk crashing the game or causing other severe issues.
 */
protected boolean canSlotsBeHidden(SlotSemantic semantic) {
    return false;
}
```